### ISSUE TITLE
Unused variable reads are not detected

### ISSUE THREAD

**Issue by herndlm** [slevomat/coding-standard#1402]
Not sure if this is an actually issue or if I'm missing something, but for the following code
```php
function foo(): void
{
    global $foo;
    $bar = true;
    $baz;
}
```
I'd expect 3 errors from `SlevomatCodingStandard.Variables.UnusedVariable.UnusedVariable`, but actually it reports only
> Unused variable $bar.

What I want to detect in the first place are unnecessary `global` statements.

I can try to give it a shot if somebody can verify that this should in fact really error as well. The current sniff reads like only writes are handled, so I might have just misunderstood it 🤔

**Comment by kukulich** [comment]
> What I want to detect in the first place are unnecessary `global` statements.

PHPStan or Psalm are better for this.

**Comment by herndlm** [comment]
Hehe, I thought a similar thing first and when it didn't work in phpstan I went looking for a sniff here :)
It would definitely be easier to implement there I guess. Thx for the input!
